### ISSUE TITLE
[Bug]: Panic: runtime error: index out of range [0] with length 0 when upgrading standalone image

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#28228]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20231102-d39ffba5-amd64 -> master-20231107-5b45a138
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus with image master-20231102-d39ffba5-amd64
2. In fact, I forgot what I did. Anyway, I ran a lot of tests.
3. Upgrade milvus image to master-20231107-5b45a138, and standalone pod panic
[standalone_pre.log](https://github.com/milvus-io/milvus/files/13277003/standalone_pre.log)

```
[2023/11/07 07:28:20.886 +00:00] [INFO] [segments/segment.go:840] ["load deleted record done"] [collectionID=445354711193951654] [partitionID=445354711193951658] [segmentID=445354711193951807] [rowNum=165] [segmentType=Growing]
[2023/11/07 07:28:20.893 +00:00] [INFO] [gc/gc_tuner.go:90] ["GC Tune done"] ["previous GOGC"=200] ["heapuse "=32] ["total memory"=254] ["next GC"=75] ["new GOGC"=200] [gc-pause=54.843µs] [gc-pause-end=1699342100892245851]
[2023/11/07 07:28:20.893 +00:00] [INFO] [segments/segment_loader.go:263] ["load segment done"] [traceID=33cfbd241d57a03343216f7f4d3f3a01] [collectionID=445354711193951654] [segmentType=Growing] [segmentID=445354711193951807]
[2023/11/07 07:28:20.893 +00:00] [DEBUG] [funcutil/parallel.go:54] [loadSegmentFunc] [total=10] ["time cost"=197.830439ms]
[2023/11/07 07:28:20.893 +00:00] [INFO] [segments/segment_loader.go:288] ["all segment load done"] [traceID=33cfbd241d57a03343216f7f4d3f3a01] [collectionID=445354711193951654] [segmentType=Growing]
panic: runtime error: index out of range [0] with length 0

goroutine 1089 [running]:
panic({0x44774c0, 0xc002a0e150})
    /usr/local/go/src/runtime/panic.go:987 +0x3bb fp=0xc0015f4470 sp=0xc0015f43b0 pc=0x1797c1b
runtime.goPanicIndex(0x0, 0x0) 
    /usr/local/go/src/runtime/panic.go:113 +0x7f fp=0xc0015f44b0 sp=0xc0015f4470 pc=0x179595f
github.com/milvus-io/milvus/internal/querynodev2/segments.(*LocalSegment).Delete(0xc001879b90, {0x679ce68, 0x0, 0x62e37c00fc31da6?}, {0x679ce68, 0x0, 0xc001fb1d00?})
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/segment.go:557 +0x712 fp=0xc0015f4588 sp=0xc0015f44b0 pc=0x3c6c172
github.com/milvus-io/milvus/internal/querynodev2/delegator.(*shardDelegator).LoadGrowing(0xc001d28fc0, {0x4c31780, 0xc001f9da70}, {0xc001fb1d00, 0xa, 0x10}, 0x179546a316f4c35f)
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/delegator/delegator_data.go:327 +0x48d fp=0xc0015f48f0 sp=0xc0015f4588 pc=0x3c8060d
github.com/milvus-io/milvus/internal/querynodev2.loadGrowingSegments({0x4c31780, 0xc001f9da70}, {0x4c690a8, 0xc001d28fc0}, 0xc001c92ea0)
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/handlers.go:114 +0x103 fp=0xc0015f4a38 sp=0xc0015f48f0 pc=0x3ca21c3
github.com/milvus-io/milvus/internal/querynodev2.(*QueryNode).WatchDmChannels(0xc0011b0b40, {0x4c31780, 0xc001f9da70}, 0xc001c92ea0)
    /go/src/github.com/milvus-io/milvus/internal/querynodev2/services.go:319 +0x1c25 fp=0xc0015f53f8 sp=0xc0015f4a38 pc=0x3cb03a5
github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).WatchDmChannels(0xc001e750a0?, {0x4c31780?, 0xc001f9da70?}, 0xc001fe6448?)
    /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:279 +0x2e fp=0xc0015f5428 sp=0xc0015f53f8 pc=0x3cc996e
github.com/milvus-io/milvus/internal/proto/querypb._QueryNode_WatchDmChannels_Handler.func1({0x4c31780, 0xc001f9da70}, {0x45bdc40?, 0xc001c92ea0})
    /go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:6302 +0x7b fp=0xc0015f5468 sp=0xc0015f5428 pc=0x25ca35b
github.com/milvus-io/milvus/pkg/util/interceptor.ServerIDValidationUnaryServerInterceptor.func1({0x4c31780, 0xc001f9da70}, {0x45bdc40, 0xc001c92ea0}, 0x7?, 0xc001b24ff0)
    /go/src/github.com/milvus-io/milvus/pkg/util/interceptor/server_id_interceptor.go:54 +0x102 fp=0xc0015f54b8 sp=0xc0015f5468 pc=0x3464742
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x4c31780?, 0xc001f9da70?}, {0x45bdc40?, 0xc001c92ea0?})
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc0015f54f8 sp=0xc0015f54b8 pc=0x23414ba
github.com/milvus-io/milvus/pkg/util/interceptor.ClusterValidationUnaryServerInterceptor.func1({0x4c31780, 0xc001f9da70}, {0x45bdc40, 0xc001c92ea0}, 0x0?, 0xc0020a09e0)
    /go/src/github.com/milvus-io/milvus/pkg/util/interceptor/cluster_interceptor.go:48 +0xf3 fp=0xc0015f5558 sp=0xc0015f54f8 pc=0x34654b3
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x4c31780?, 0xc001f9da70?}, {0x45bdc40?, 0xc001c92ea0?})
```

**Comment by yah01** [comment]
This would be fixed by #28213

**Comment by yanliang567** [comment]
@yah01 please assign back to us as the pr merged. Thx
/assign @yah01 
/unassign

**Comment by ThreadDao** [comment]
Fixed '[master-20231108-ecb3f585](https://hub.docker.com/layers/milvusdb/milvus/master-20231108-ecb3f585/images/sha256-02c9ab6a7499e71b4d0e421b42dd8dfc7c5c15cf40f36193a3540d181c2f47b3?context=explore)'
